Deprecate config item `max-memory` and add items `server-memory-quota` and `memory-usage-alarm-ratio` (#4706)
pingcap/docs-cn: tidb-configuration-file.md

+ 默认值：0
+ 默认值为 0 表示使用机器上所有的 CPU；如果设置成 n，那么 TiDB 会使用 n 个 CPU 数量。

### `server-memory-quota`

> + TiDB 内存使用限制。
> + 默认值：0 （Byte）

Reviewer: ```suggestion
+ TiDB 内存使用限制，单位为字节。
+ 默认值：0 
```

+ 默认值为 0 表示无内存限制。

### `memory-usage-alarm-ratio`

+ TiDB 内存使用报警阈值。
+ 默认值：0.8

> + 该配置项的有效范围为 0 到 1。 如果配置该选项为 0 或 1，则表示关闭内存阈值报警功能。否则，当 TiDB 检测到内存使用超过了阈值，则会将相关信息记录到目录 `tmp-storage-path/record` 中。

Reviewer 2: 1. 这个太简单了，得写下会记录哪些信息。
2. 此外，关于tmp-storage-path 得加个链接到对应的文档项上去
3. server-memory-quota 也需要加个链接到对应的文档项上
4. system memory size 换成中文

> + 该配置项的有效范围为 0 到 1。 如果配置该选项为 0 或 1，则表示关闭内存阈值报警功能。否则，当 TiDB 检测到内存使用超过了阈值，我们认为 TiDB 目前存在 OOM 的风险，会将当前正在执行的所有 SQLs 中内存使用最高的 10 条 SQL 和运行时间最长的 10 条 SQL 以及 heap profile 记录到目录 [`tmp-storage-path/record`](/tidb-configuration-file.md#tmp-storage-path) 中,并输出一条包含关键子 `the Tidb instance has the risk of OOM` 的日志。

Reviewer: ```suggestion
+ 该配置项的有效范围为 0 到 1。如果配置该选项为 0 或 1，则表示关闭内存阈值报警功能。否则，当 TiDB 检测到内存使用超过了阈值，则会认为存在 OOM 的风险，会将当前正在执行的所有 SQL 语句中内存使用最高的 10 条语句和运行时间最长的 10 条语句以及 heap profile 记录到目录 [`tmp-storage-path/record`](/tidb-configuration-file.md#tmp-storage-path) 中，并输出一条包含关键字 `the Tidb instance has the risk of OOM` 的日志。
```

> + 注意：如果配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota) 被设置且大于 0，则内存报警阈值将为 `memory-usage-alarm-ratio * server-memory-quota`；否则，内存报警阈值将为 `memory-usage-alarm-ratio * 系统内存大小`。

Reviewer: ```suggestion
```

> + 注意：如果配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota) 被设置且大于 0，则内存报警阈值将为 `memory-usage-alarm-ratio * server-memory-quota`；否则，内存报警阈值将为 `memory-usage-alarm-ratio * 系统内存大小`。

Reviewer 3: 此处提到内存报警阈值为 `memory-usage-alarm-ratio` 与 `server-memory-quota` 两个配置项的值的乘积，但第 294 行对 `memory-usage-alarm-ratio` 这个配置项的解释也是“TiDB 内存使用报警阈值”，前后有点矛盾。而从配置项名称中的“ratio”来看，应该这个配置项的值规定的是内存使用达到何种比例时报警，是否考虑修改一下第 294 行对配置项的说明？

### `memory-usage-alarm-ratio`

> + TiDB 内存使用报警阈值占总内存的比例。

Reviewer: ```suggestion
+ TiDB 内存使用占总内存的比例超过一定阈值时会报警。该配置项的有效范围为 0 到 1。如果配置该选项为 0 或 1，则表示关闭内存阈值报警功能。
```

+ 默认值：0.8

> + 该配置项的有效范围为 0 到 1。如果配置该选项为 0 或 1，则表示关闭内存阈值报警功能。否则，当 TiDB 检测到内存使用超过了阈值，则会认为存在 OOM 的风险，会将当前正在执行的所有 SQL 语句中内存使用最高的 10 条语句和运行时间最长的 10 条语句以及 heap profile 记录到目录 [`tmp-storage-path/record`](/tidb-configuration-file.md#tmp-storage-path) 中，并输出一条包含关键字 `tidb-server has the risk of OOM` 的日志。

Reviewer: ```suggestion
+ 当内存阈值报警功能开启时，如果配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota) 未设置，则内存报警阈值为 `memory-usage-alarm-ratio * 系统内存大小`；如果 `server-memory-quota` 被设置且大于 0，则内存报警阈值为 `memory-usage-alarm-ratio * server-memory-quota`。
+ 当 TiDB 检测到内存使用超过了阈值，则会认为存在 OOM 的风险，会将当前正在执行的所有 SQL 语句中内存使用最高的 10 条语句和运行时间最长的 10 条语句以及 heap profile 记录到目录 [`tmp-storage-path/record`](/tidb-configuration-file.md#tmp-storage-path) 中，并输出一条包含关键字 `tidb-server has the risk of OOM` 的日志。
```

> + 注意：如果配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota) 被设置且大于 0，则内存报警阈值将为 `memory-usage-alarm-ratio * server-memory-quota`；否则，内存报警阈值将为 `memory-usage-alarm-ratio * 系统内存大小`。

Reviewer: ```suggestion
```
